functional/FunctionalTest.java

```diff
@@ -394,105 +418,79 @@ public static void putObject_test2() throws Exception {
    * Test: With content-type: putObject(String bucketName, String objectName, String filename, String contentType).
    */
   public static void putObject_test3() throws Exception {
-    
     if (!mintEnv) {
       System.out.println("Test: putObject(String bucketName, String objectName, String filename,"
                         + " String contentType)");
     }
 
     long startTime = System.currentTimeMillis();
-    String filename = createFile1Mb();
-    boolean success = true;
-
     try {
+      String filename = createFile1Mb();
       client.putObject(bucketName, filename, filename, customContentType);
       Files.delete(Paths.get(filename));
       ObjectStat objectStat = client.statObject(bucketName, filename);
-      success = customContentType.equals(objectStat.contentType());
-      client.removeObject(bucketName, filename);
-
-      if (success) {
-        mintSuccessLog("putObject(String bucketName, String objectName, String filename, String contentType)", 
-                      "contentType: " + customContentType, startTime);
-      } else {
-        mintFailedLog("putObject(String bucketName, String objectName, String filename, String contentType)", 
-                      "contentType: " + customContentType, startTime, null, null);
+      if (!customContentType.equals(objectStat.contentType())) {
+        throw new Exception("content type mismatch, expected: " + customContentType + ", got: "
```

Comment:
Where will removeObject get called, after the exception is thrown?

Reply:
No. When no exception is thrown, below code is executed